### PR TITLE
fix(dtfs2-none): fix boolean coersion for sql config

### PR DESCRIPTION
**Pull request by AlexBramhill**
## Introduction :pencil2:
There is a surprising way that zod coerces booleans (https://zod.dev/?id=coercion-for-primitives). This means that `"false"` would parse as `true`

## Resolution :heavy_check_mark:
I've made the coercion a lot more restrictive. We could have simply checked and done `typeof val === 'string' && val.toLowerCase() === 'false' ? false : val`, however I'm not sure we want to coerse all things truthy and falsy.

### CODE REVIEW

File: libs/common/src/helpers/schemas/z-boolean-coerce.test.ts [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  describe('when parsing a value with zBooleanCoerce', () => {
+    describe('when the provided value can be coerced to a boolean', () => {
+      it.each(getSuccessTestCases())('should return $result when the provided value is $description', ({ value, result }) => {
+        expect(zBooleanCoerce.parse(value)).toBe(result);
+      });
+    });
+
+    describe('when the provided value cannot be coerced to a boolean', () => {
+      it.each(getFailureTestCases())('should throw an error when the provided value is $description', ({ value }) => {
+        expect(() => zBooleanCoerce.parse(value)).toThrow();
```
Comment (MarRobSoftwire) [inline libs/common/src/helpers/schemas/z-boolean-coerce.test.ts:13]: fwiw could ensure we're not getting an error for another reason
```suggestion
        expect(() => zBooleanCoerce.parse(value)).toThrow(ZodError);
```

Comment (MarRobSoftwire) [inline libs/common/src/helpers/schemas/z-boolean-coerce.test.ts:13]: fwiw could ensure we're not getting an error for another reason
```suggestion
        expect(() => zBooleanCoerce.parse(value)).toThrow(ZodError);
```

**Review by MarRobSoftwire — APPROVED** [review]

File: libs/common/src/helpers/schema/z-boolean-strict-coerce.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import z from 'zod';
+
+const booleanLike = z.union([
```
Comment (abhi-markan) [inline libs/common/src/helpers/schema/z-boolean-strict-coerce.ts:3]: Do we even need `zod` package, can we not do this natively?

**Review by abhi-markan — COMMENTED** [review]
Good find and solution, however can we achieve this natively wihtout using `zod`?

File: libs/common/src/helpers/schema/z-boolean-strict-coerce.ts (hunk 2 above)
Comment (AlexBramhill) [reply libs/common/src/helpers/schema/z-boolean-strict-coerce.ts:3]: I'd say zod offers a lot more than us hand rolling validation ourselves (our existing validation has a few blind spots -- which zod covers nicely).

It also provides good error msg reporting, and is easy to use and read.

It also allows for type inference from schemas, meaning once we add validation to something, we also can generate a typescript type from it (one source of truth):
```
const settingsSchema = z.object({
  darkMode: z.boolean().default(false),
  notifications: z.boolean().default(true),
});

type Settings = z.infer<typeof settingsSchema>;
```

tldr - I think it offers a massive timesave into us rolling our own validation, with some great benefits like type inference.

File: libs/common/src/helpers/schema/z-boolean-strict-coerce.ts [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+ * This addresses an issue where 'false' would be coerced to true -- causing issues in env vars.
+ */
+export const zBooleanStrictCoerce = booleanLike.transform((val) => {
+  if (typeof val === 'string') {
+    return ['true', '1'].includes(val.toLowerCase());
+  }
+  if (typeof val === 'number') {
+    return val === 1;
+  }
+  return val;
```
Comment (MarRobSoftwire) [inline libs/common/src/helpers/schema/z-boolean-strict-coerce.ts:23]: fwiw I think we can cast the value to string, since we've reduced it to one of the above literals
```suggestion
return ['true', '1'].includes(String(val).toLowerCase());
```

**Review by MarRobSoftwire — APPROVED** [review]
Another minor comment but LGTM

File: libs/common/src/helpers/schema/z-boolean-strict-coerce.ts (hunk 3 above)
Comment (AlexBramhill) [reply libs/common/src/helpers/schema/z-boolean-strict-coerce.ts:23]: great spot
